src/pymor/discretizations/iosys.py

```diff
@@ -0,0 +1,1702 @@
+        return True
+    if (isinstance(operator, BlockDiagonalOperator) and
+            all(_is_like_identity_operator(op) for op in operator._blocks.diagonal())):
+        return True
+    return False
+
+
+class InputOutputSystem(DiscretizationBase):
+    """Base class for input-output systems."""
+
+    def __init__(self, m, p, cont_time=True, cache_region='memory', name=None, **kwargs):
+        self.m = m
+        self.p = p
+        super().__init__(cache_region=cache_region, name=name, **kwargs)
+        self.cont_time = cont_time
+
+    def _solve(self, mu=None):
+        raise NotImplementedError
+
+    def __add__(self, other):
+        """Add two input-state-output systems."""
+        assert type(self) == type(other)
+        assert self.cont_time == other.cont_time
+
+        def add_operator(op, other_op):
+            if op.source.id == 'INPUT':
+                if op.range.id == 'OUTPUT':
+                    return BlockOperator([[(op + other_op).assemble()]], source_id=op.source.id, range_id=op.range.id)
+                elif op.range.id == 'STATE':
+                    return BlockOperator.vstack((op, other_op), source_id=op.source.id, range_id=op.range.id)
+                else:
+                    raise NotImplementedError
+            else:
+                if op.range.id == 'OUTPUT':
+                    return BlockOperator.hstack((op, other_op), source_id=op.source.id, range_id=op.range.id)
+                elif op.range.id == 'STATE':
+                    return BlockDiagonalOperator((op, other_op), source_id=op.source.id, range_id=op.range.id)
+                else:
+                    raise NotImplementedError
+
+        new_operators = {k: add_operator(self.operators[k], other.operators[k]) for k in self.operators}
+
+        return self.with_(operators=new_operators, cont_time=self.cont_time)
+
+    def __neg__(self):
+        """Negate input-state-output system."""
+        new_operators = {k: (op * (-1)).assemble() if op.range.id == 'OUTPUT' else op
+                         for k, op in self.operators.items()}
+
+        return self.with_(operators=new_operators)
+
+    def __sub__(self, other):
+        """Subtract two input-state-output system."""
+        return self + (-other)
+
+    def eval_tf(self, s, mu=None):
+        """Evaluate the transfer function."""
+        raise NotImplementedError
+
+    def eval_dtf(self, s, mu=None):
+        """Evaluate the derivative of the transfer function."""
+        raise NotImplementedError
+
+    @cached
+    def bode(self, w):
+        """Evaluate the transfer function on the imaginary axis.
+
+        Parameters
+        ----------
+        w
+            Frequencies at which to compute the transfer function.
+
+        Returns
+        -------
+        tfw
+            Transfer function values at frequencies in `w`,
+            |NumPy array| of shape `(self.p, self.m, len(w))`.
+        """
+        if not self.cont_time:
+            raise NotImplementedError
```

Comment:
Probably a stupid question, but shouldn't be the following code the same for the contiuous time case?

Reply:
Unfortunately, no. For the continuous-time case, evaluation of the transfer function is done over the imaginary axis, but for the discrete-time case, it is over the unit circle.

Reply:
Oh, sorry, somehow had an additional negation in my head.